sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixConfiguration.scala

```diff
@@ -36,4 +38,17 @@ class KalixConfiguration(applicationContext: ApplicationContext) {
     kalix
   }
 
+  @Component
+  class KalixComponentInjectionBlocker extends BeanPostProcessor {
+    override def postProcessBeforeInitialization(bean: AnyRef, beanName: String): AnyRef = {
+      if (KalixServer.kalixComponents.exists(_.isAssignableFrom(bean.getClass)))
+        throw new IllegalArgumentException(
+          s"${bean.getClass.getName} is Kalix component and is marked as a Spring bean for automatic wiring. " +
+          "Kalix components cannot be accessed directly and therefore cannot be wired into other classes. " +
+          "In order to interact with a Kalix component, you should call it using the provided KalixClient.")
+
+      bean
+    }
+  }
```

Comment:
This is a better way of block wiring of Kalix components. In my previous commit (df7acee24699394021c53cd2aa5fd271361118d7), I was trying to simply scan the component constructor to look for param that were also Kalix components. 

That was flawed. One could by-pass the check by wrapping a Kalix component into another class, like in:

```scala
class AnotnerAction extends Action
class SomeComponent(action: AnotnerAction)
class OneAction(someComponent: SomeComponent) extends Action
```

When checking the constructor of `OneAction` we won't find `AnotnerAction`, we will only see `SomeComponent`

With a `BeanPostProcessor`, we can throw an exception when the user mark any Kalix Components with a Spring stereotype. Btw, without using sterotype, Spring can't wire them anyway.